### ISSUE TITLE
Introduce ephemeral remotes to use rclone without configuration

### ISSUE THREAD

**Issue by ivandeex** [rclone/rclone#4720]
#### What is your current rclone version (output from `rclone version`)?

1.53.2 (latest)

#### What problem are you are trying to solve?

TL;DR I want to run this straight away:
```
rclone ls :sftp:subdir --sftp-host=8.8.8.8
```

Let me cite https://github.com/restic/restic/issues/2411#issuecomment-587722167

> > _[rephrased]_ I suggest phasing out all restic backends in favor of rclone

> The one thing I don't like about rclone is that I have to configure it before running it, instead of just being able to give it two URIs as the source and target.

#### How do you think rclone should be changed to solve that?

The [:memory:](https://rclone.org/memory/) backend has been explicitly allowing to use it without config file for ages (rclone will just [create empty one](https://github.com/rclone/rclone/blob/v1.53.2/fs/config/config.go#L222) internally)

I suggest to introduce _emphemeral remotes_ in rclone by extending the _first-colon_ syntax to all backend types so that one can do like
```
rclone copy ./file.txt :ftp:some/dir --ftp-host=8.8.8.8
```
Of course one could also extend this with 12-factor settings:
```
export RCLONE_FTP_USER=smith
```

An implementation would create an _ephemeral fs_ similar to [temporary fs](https://github.com/rclone/rclone/blob/v1.53.2/fs/fs.go#L1352) with a new [config getter](https://github.com/rclone/rclone/blob/v1.53.2/fs/fs.go#L1282) whenever `:backendType:` is found on command line. The getter would create an ephemeral config section (and optionally fill it from env or cli args).

Probably `rclone ls yandex://subdir` syntax (without first colon and with obligatory double-slash after backend type) might work too (in contrast to `yandexRemote:subdir` for permanent remotes), but I'm not sure.
Just an idea.

**Comment by ncw** [comment]
> TL;DR I want to run this straight away:
> 
> ```
> rclone ls :sftp:subdir --sftp-host=8.8.8.8
> ```

Doesn't that work already? Or am I missing something?

**Comment by ivandeex** [comment]
Sorry, it does work. I missed that in the docs.

**Comment by ncw** [comment]
No prob!
